src/openrct2/ride/RideRatings.cpp

```diff
@@ -207,6 +207,8 @@ static void ride_ratings_update_state_2()
     int32_t trackType = gRideRatingsCalcData.proximity_track_type;
 
     TileElement* tileElement = map_get_first_element_at(x, y);
+    if (tileElement == nullptr)
+        return;
```

Comment:
will need to set `gRideRatingsCalcData.state = RIDE_RATINGS_STATE_FIND_NEXT_RIDE;` prior to returning.